### ISSUE TITLE
Feature Request: Ring Security System Support

### ISSUE THREAD

**Issue by adam-fox** [tchellomello/python-ring-doorbell#104]
Would love to see support for the new Ring Security System (https://shop.ring.com/pages/security-system). Not sure if this is helpful but I came across this in my searching: https://github.com/davglass/doorbot/issues/26

Thanks for all the work you’ve done on this and all the Home Assistant components.

**Comment by tchellomello** [comment]
Hello @adam-fox, 
   
    I don't have plans to implement it myself as I don't own a Ring Security System but any contributions to add support to it would be very welcome. 

    Thank you for your kind words. :+1: 

mmello

**Comment by litobro** [comment]
I'm receiving my Ring Alarm system in the next few days in the mail. I'm willing to take a stab at trying to put in a simple implementation. Is there any best practices or gotchas we should be aware about?

I'll admit my interest in getting this working is in using this as a component for home-assistant.

**Comment by joshfokis** [comment]
I am also working on this although having issues monitoring the traffic with the MITM proxy. I it works under the same call for the doorbots/{id}/status. I did get the id from using the query call.

**Comment by litobro** [comment]
My quick attempt to MITM proxy my Ring app shows connection errors - indicative that SSL pinning is being used to prevent intercepting traffic. I'm on an Android P device running the latest version of the app. This would explain why @joshfokis is having difficulty intercepting the traffic.

I don't currently have access to a rooted device to bypass the pinning or the ability to recompile the app without the pinning in place. I'll attempt to use the other repos linked to try and port rather than detect some of that data. 

@tchellomello did you originally develop this using interception? Were you able to circumvent the SSL pinning?

**Comment by shadowbq** [comment]
MITM worked originally with the RING app, they only recently in the past few months pinned the SSL certificate. I was able to MITM to identify the snooze/quiet etc. functions when they added them

**Comment by joshfokis** [comment]
I was hoping to be consistent and use the API calls and not the websockets that have been used. Only thing I have found out is when calling ring_devices it lists it under base_stations but you get the health call from doorbots but you cannot call history from it.

**Comment by litobro** [comment]
I agree, I had hoped to use the API calls as well, though based on inspecting the traffic in the web browser, and making some HTTP requests to the API, I can only get the same information as you. Based on the behaviour of the app (it takes additional time to 'connect' to the base station and load), it looks like the new app may be using websockets for that portion as well. Are we sure that there's API exposed for alarms at all?

**Comment by litobro** [comment]
After some experimentation, the authentication from the API is not sufficient authentication for the websockets. This is irritating as it will require quite a substantive rewrite to implement some of the websockets authentication and other code.

**Comment by toscano** [comment]
I was going to take a look myself. Perhaps this helps with the auth issues [homespun/ring-alarm](https://github.com/homespun/ring-alarm)

**Comment by dshokouhi** [comment]
I found this as well, maybe it will be helpful? 

https://github.com/acolytec3/ring-alarm-HA-plugin

**Comment by Pixolsc2** [comment]
(nearly 3 years later) For anyone still interested, see #250
